pkg/disttask/framework/metering/metering.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metering
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/metering_sdk/common"
+	mconfig "github.com/pingcap/metering_sdk/config"
+	"github.com/pingcap/metering_sdk/storage"
+	meteringwriter "github.com/pingcap/metering_sdk/writer/metering"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/util/logutil"
+	"go.uber.org/zap"
+)
+
+const (
+	writeInterval = 60
+	// The timeout can not be too long because the pod grace termination period is fixed.
+	writeTimeout = 10 * time.Second
+	category     = "disttask"
```

Comment:
```suggestion
	dxfCategory     = "dxf"
```
can we use `dxf`, disttask is the old naming